etc/webapi.xml

```diff
@@ -0,0 +1,12 @@
+<?xml version="1.0"?>
+<routes xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+        xsi:noNamespaceSchemaLocation="urn:magento:module:Magento_Webapi:etc/webapi.xsd">
+
+    <!-- Subscriptions Service -->
+    <route url="/V1/paytrail/subscription/cancel/:subscription_id" method="GET">
+        <service class="Paytrail\PaymentService\Api\SubscriptionManagementInterface" method="cancelSubscription" />
+        <resources>
+            <resource ref="anonymous"/>
```

Comment:
Incorrect resource declaration. This resource needs to be restricted to customer only. Right now it's open to everyone. 

See vendor/magento/module-customer/etc/webapi.xml for example:

```
    <route url="/V1/customers/me/password" method="PUT">
        <service class="Magento\Customer\Api\AccountManagementInterface" method="changePasswordById"/>
        <resources>
            <resource ref="self"/>
        </resources>
        <data>
            <parameter name="customerId" force="true">%customer_id%</parameter>
        </data>
    </route>
```